stockroom/storages/modelstore.py

```diff
@@ -5,93 +5,130 @@
+        if len(weights) != numLayers:
+            raise RuntimeError(
+                "Critical: Discrepancy in the stored data: numLayers != len(weights)")
         if library == 'torch':
-            str_layers = co.metadata[parser.metakey(name, 'layers')]
-            layers = parser.string_to_layers(str_layers)
-            if len(layers) != num_layers:
-                raise RuntimeError("Critical: length doesn't match. Raise an issue")
-            state = {layers[i]: torch.from_numpy(weights[i]) for i in range(num_layers)}
+            import torch
+            state = {layers[i]: torch.from_numpy(weights[i]) for i in range(numLayers)}
             model.load_state_dict(state)
         else:
             model.set_weights(weights)
 
+    @staticmethod
+    def store_metadata(co, name, library, longest, dtypes, numLayers, layers=None):
+        co.metadata[parser.metakey(name, 'library')] = library
+        if library == 'torch':
+            if layers is None:
+                raise ValueError("Could not fetch layer information for the torch model")
+            co.metadata[parser.metakey(name, 'layers')] = parser.encode(layers)
+        co.metadata[parser.metakey(name, 'library')] = library
+        co.metadata[parser.metakey(name, 'longest')] = longest
+        co.metadata[parser.metakey(name, 'numLayers')] = numLayers
+        co.metadata[parser.metakey(name, 'dtypes')] = parser.encode(dtypes)
 
+    @staticmethod
+    def get_metadata(co, name):
+        library = co.metadata[parser.metakey(name, 'library')]
+        if library == 'torch':
+            layers = co.metadata[parser.metakey(name, 'layers')]
+            layers = parser.decode(layers)
+        else:
+            layers = None
+        library = co.metadata[parser.metakey(name, 'library')]
+        longest = co.metadata[parser.metakey(name, 'longest')]
+        numLayers = co.metadata[parser.metakey(name, 'numLayers')]
+        dtypes = co.metadata[parser.metakey(name, 'dtypes')]
+        dtypes = parser.decode(dtypes)
+        return library, longest, dtypes, int(numLayers), layers
+
+    def get_arrayset(self, co, name):
+        try:
+            aset = co.arraysets[name]
+            aset.__enter__()
+            self.opened_aset.append(aset)
+            return aset
+        except KeyError:
+            return None
+
+    def create_arrayset(self, co, name, longest=None, dtype=None, variable_shape=False):
+        self.close_arraysets()
+        aset = co.arraysets.init_arrayset(
+            name, dtype=np.dtype(dtype), shape=(longest,), variable_shape=variable_shape)
+        self.opened_aset.append(aset)
+
+        # open all the closed arraysets
+        for val in self.opened_aset:
+            val.__enter__()
+        return aset
 
+    def close_arraysets(self):
+        for aset in self.opened_aset:
+            aset.__exit__()
 
 
+def modelstore(write=False):
+    return ModelStore(write=write)
```

Comment:
Why does this exist? If you need alternative constructors, use `@classmethod`. See hangar `Arrayset` methods here:

https://github.com/tensorwerk/hangar-py/blob/519d63b278a35367a89dbb9f49bd39f29c725e81/src/hangar/arrayset.py#L1448-L1541